src/dev/flang/ast/InlineArray.java

```diff
@@ -277,7 +277,7 @@ public void checkTypes()
 
     for (var e : _elements)
       {
-        if (!elementType.isAssignableFrom(e.type()))
+        if (!elementType.isDirectlyAssignableFrom(e.type()))
```

Comment:
I think this would have the effect that code like
```
  x array (option i32) := [nil, 42, nil, 4711]
```
would no longer work or does it?

Reply:
It should still work. (I just tested it.) 
Assumption is that checkTypes happens after boxing.

Reply:
Yes, did not check this well. During checkTypes, isAssignableFrom should no longer be used.